### PR TITLE
fix: #9061 secret migration could fail and --yes was not working

### LINKED ISSUE #9061: After Updating to Amplify CLI 7.5.0 OAuth problems
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.16.0

### Amplify CLI Version

7.5.0

### What operating system are you using?

Mac

### Amplify Categories

auth

### Amplify Commands

add

### Describe the bug

After updating AWS Cli to 7.5.0 when I do `amplify add auth`

Then using "Default configuration with Social Provider (Federation)" after inputting Facebook App ID and Secret it shows

`Cannot read property 'filter' of undefined
TypeError: Cannot read property 'filter' of undefined
    at mergeDeploymentSecrets (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/deploymentSecretsHelper.ts:13:57)
    at AmplifyToolkit.saveEnvResourceParameters [as _saveEnvResourceParameters] (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/envResourceParams.ts:81:31)
    at /Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/handlers/resource-handlers.ts:59:21
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Object.executeAmplifyCommand (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/index.js:421:3)
    at executePluginModuleCommand (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:204:3)
    at executeCommand (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:30:5)
    at Object.run (/Users/gianfrancodelmundo/.nvm/versions/node/v14.16.0/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5)`

### Expected behavior

Should Create Auth with Social Federation

### Reproduction steps

1. amplify init
2. amplify add auth
3. Select Default Configuration with Social Provider
4. Select Facebook/Google
5. Add App ID or Secret

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
2021-11-24T01:39:55.305Z|info : amplify add auth  
2021-11-24T01:40:03.141Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-11-24T01:41:34.967Z|info : amplify push core  
2021-11-24T01:41:41.084Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-11-24T01:41:41.371Z|info : amplify-provider-awscloudformation.aws-s3.deleteDirectory.s3.getAllObjectVersions([{"BucketName":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:41.372Z|info : amplify-provider-awscloudformation.aws-s3.getAllObjectKey.s3.listObjectVersions([{"Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment","Prefix":"hooks/"}])
2021-11-24T01:41:42.070Z|info : amplify-provider-awscloudformation.aws-s3.deleteAllObjects.s3.deleteObjects (0 of 1)([{"Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:42.518Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ks/[***]st-[***]sample","Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:43.022Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ks/[***]re-[***]sample","Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:43.324Z|info : amplify-provider-awscloudformation.aws-s3.uploadFile.s3.putObject([{"Key":"[***]ks/[***]E.md","Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:43.831Z|info : amplify-provider-awscloudformation.zip-util.downloadZip.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip"},null])
2021-11-24T01:41:43.833Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])
2021-11-24T01:41:44.233Z|info : amplify-provider-awscloudformation.initialize-env.run.cfn.updateamplifyMetaFileWithStackOutputs([{"StackName":"[***]ify-[***]atest-[***]ev-[***]53"}])
2021-11-24T01:41:44.235Z|info : amplify-provider-awscloudformation.aws-cfn.updateamplifyMetaFileWithStackOutputs.cfn.describeStackResources([{"StackName":"[***]ify-[***]atest-[***]ev-[***]53"}])
2021-11-24T01:41:44.498Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]ify-[***]atest-[***]ev-[***]53"}])
2021-11-24T01:41:45.048Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]ify-[***]json","Bucket":"[***]ify-[***]atest-[***]ev-[***]53-[***]ment"}])


```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by attilah**
#### Description of changes

Fix an issue when secret migration could fail when `~/.aws/amplify/deployment-secrets.json` does not have an `appSecrets` property persisted.
Fix the `--yes` flag usage as `exeInfo` is not initialized when this prompt was executing.

#### Issue #, if available

Fixes #9061

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/utils/headless-input-utils.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import readline from 'readline';
+import _ from 'lodash';
```
Comment (cjihrig) [inline packages/amplify-cli/src/utils/headless-input-utils.ts:2]: This looks unused.

**Review by cjihrig — APPROVED** [review]
One small comment. LGTM

File: packages/amplify-cli/src/utils/headless-input-utils.ts (hunk 1 above)
Comment (attilah) [reply packages/amplify-cli/src/utils/headless-input-utils.ts]: Yeah refactored my changes to not need it. Removed, thx.

**Review by jhockett — COMMENTED** [review]
Changes LGTM, is there an e2e test that covers an env checkout with secrets present in the project?

**Review by jhockett — APPROVED** [review]

**Review by ammarkarachi — APPROVED** [review]

**Pull request merged**